### PR TITLE
Backup function for database

### PR DESCRIPTION
**Pull request by julianfortune**
- Lets you backup the database to a `.json` file.
- The max size for this kinda backup is like 4mb so I don't think we need to worry
- Could be helpful once we hookup an api for adding quizzes

### CODE REVIEW

**Review by merezal — APPROVED** [review]

File: server.js [hunk 1]
```diff
@@ -140,6 +142,32 @@ async function lookupRecentQuizzes(count, completion) {
 
 ///SECTION: DB utility functions
 
+// Backs up database to a JSON file with name backup<date>.json.
+// ⚠️ Don't run in nodemon or it will cycle forever!
+function backupDatabaseToJSON() {
+    console.log("💾  Backing up database...");
+
+    database.collection(quizCollection).find({}).toArray(function(err, allQuizzes) {
```
Comment (merezal) [inline server.js:183]: Can't run the server I'm getting this error:

TypeError: Cannot read property 'collection' of undefined

line 150

**Review by merezal — CHANGES_REQUESTED** [review]

File: server.js (hunk 1 above)
Comment (julianfortune) [reply server.js:183]: @MichaelJereza Could you tell me what steps you took that caused this issue? Sounds like the database didn't get set up properly, but I can't think why... 🧐

Comment (julianfortune) [reply server.js:183]: i added an if statement that can be set to `true` to run the backup. Maybe give it a try now?

File: server.js [hunk 2]
```diff
@@ -134,6 +134,9 @@ MongoClient.connect(mongoDBURL, function(err, client) {
 
     // Change this to `true` to clear the db and seed fresh from json.
     if (false) seedDatabaseFromJSON('./exampleQuizzes.json')
+
+    // Change this to backup db. Do not use nodemon.
+    if (false) backupDatabaseToJSON()
```
Comment (julianfortune) [inline server.js:139]: This lil guy here ^

**Review by merezal — APPROVED** [review]
Very nice

**Pull request merged**